### ISSUE TITLE
requests throwing EPROTO

### ISSUE THREAD

**Issue by safebyte** [ondreian/ebay-promised#12]
It seems like ebay is having some certificate issues which are causing the module to crash the whole process with:

```
events.js:154
      throw er; // Unhandled 'error' event
      ^

Error: write EPROTO 139970400741248:error:1408F10B:SSL routines:SSL3_GET_RECORD:
       wrong version number:../deps/openssl/openssl/ssl/s3_pkt.c:362:
    at exports._errnoException (util.js:856:11)
    at WriteWrap.afterWrite (net.js:763:14)
```

The problem is also described here: https://github.com/nodejs/node/issues/3692
As supposed by @kevinburkeshyp a workaround is to pass these options to the `request` object:

```
agentOptions: {
      ciphers: 'ALL',
      secureProtocol: 'TLSv1_method',
}
```

I've tried to pass them to `request-promised` but failed, how can this be fixed?
Thanks!

**Comment by ondreian** [comment]
Thanks for bringing this to my attention, I am unsure why I have yet to see this crop up during any of the newly added functional tests, I must have gotten lucky and missed their TLS v1.0 servers in the sandbox or something.

What node version are you running for reference?

I will see about finding a resolution tonight or tomorrow morning.

**Comment by safebyte** [comment]
I've been working with `1.5.0` - and I should've updated before posting this. Because after upgrading to `2.0.0` the first tests seem to work fine. I'll keep you updated, thank you and sorry for the inconvenience!

**Comment by ondreian** [comment]
@safebyte no problem. 

I should probably add the "engines" property to the package.json file to ensure people are notified about `ebay-promised` wanting one of the newer versions.

**Comment by safebyte** [comment]
After sending approx. 600 calls to eBay using `2.0.0` I've received the error again:

```
events.js:154
      throw er; // Unhandled 'error' event
      ^

Error: write EPROTO 139893095102336:error:1408F10B:SSL routines
:SSL3_GET_RECORD:wrong version number:../deps/openssl/openssl/ssl/s3_pkt.c:362:
at exports._errnoException (util.js:856:11)
at WriteWrap.afterWrite (net.js:763:14)
```

But it seems to happen a lot less frequent than with `1.5.0`

**Comment by ondreian** [comment]
Can you try upgrading to 6.0.0 and see if you still encounter the same problems?  

I am having trouble reproducing this bug even on 2.0.0

**Comment by safebyte** [comment]
Can you give me a hint? The newest version npm is giving me is `2.0.0`:

```
λ npm install ebay-promised --save
npm WARN package.json qs@6.2.0 No README data
ebay-promised@2.0.0 node_modules\ebay-promised
```

**Comment by ondreian** [comment]
Oh, I mean your Node.js version, not the version of `ebay-promised`.

when you type `node -v` what version are you seeing?

**Comment by safebyte** [comment]
Oh okay, I'm on the latest stable now (6.2.1) and will keep you updated (test are running).

**Comment by safebyte** [comment]
And here we go again.. Same error as mentioned before. Very hard (impossible) to reproduce on purpose. I think it depends on the server eBay is using to process the request. Still happens a lot less frequent than before (approx. every 500-700 reqs).
I am using `ReviseInventoryStatus` && `InventoryStatus`

**Comment by kevinburkeshyp** [comment]
It definitely depends on the eBay server, you have to hit a TLS 1.2 node and then hit a TLS 1.0 node. I had to continually restart the node process to get it to work, I'd run a process for maybe 1500 requests and then quit it and try a new one.

Fortunately you should be able to expose the issue by hitting the root, you don't need to make expensive API calls to trigger this.

-kevin

**Comment by ondreian** [comment]
@safebyte I have applied your suggested patch and published as `2.1.0`

Try out the newest version and see if it doesn't fix the issue for now.

@kevinburkeshyp thanks for your gist, it helped immensely with reproducing
